### PR TITLE
Features with containers

### PR DESCRIPTION
**Pull request by MrBooAR**
adding custom hashset and arraylist and test them

### CODE REVIEW

File: src/Containers/CustomArrayList.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package Containers;
+
+import java.util.Arrays;
+
+public class CustomArrayList<T> {
+    private Object[] elements;
+    private int size;
+
+    public CustomArrayList() {
+        elements = new Object[10]; // Initial capacity
```
Comment (Darya-Drutsko) [inline src/Containers/CustomArrayList.java]: don't use magic numbers like 10 here, instead add field default capacity with this value

File: src/Containers/CustomArrayList.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    private void ensureCapacity() {
+        if (size == elements.length) {
+            elements = Arrays.copyOf(elements, elements.length * 2); // Double the capacity
+        }
+    }
+
+    @Override
+    public String toString() {
+        return Arrays.toString(Arrays.copyOf(elements, size));
```
Comment (Darya-Drutsko) [inline src/Containers/CustomArrayList.java:48]: why do we need here copy of?

File: src/Containers/CustomArrayList.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    @SuppressWarnings("unchecked")
+    public T getbyindex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        return (T) elements[index];
+    }
+
+    public void deletebyindex(int index) {
```
Comment (Darya-Drutsko) [inline src/Containers/CustomArrayList.java]: use camel case for method names

File: src/Containers/CustomArrayList.java (hunk 2 above)
Comment (MrBooAR) [reply src/Containers/CustomArrayList.java:48]: i use Arrays.copyOf in toString to only include the actual elements up to the current size

**Pull request merged**